sim/battle.ts

```diff
@@ -1228,10 +1235,14 @@ export class Battle extends Dex.ModdedDex {
 		if (this.ended) {
 			return false;
 		}
-		if (side === 'p1' || side === 'p2') {
-			side = this[side];
-		} else if (side !== this.p1 && side !== this.p2) {
+		if (side instanceof Side && !this.sides.includes(side)) {
```

Comment:
PS generally doesn't use `instanceof` in case of duck typing. What was wrong with the previous approach of just putting `typeof` first?

Reply:
Actually what's with this code in the first place? What's it trying to prevent?

Reply:
I dunno, there was already a check in place to look for an invalid `Side` object, so I kept it.

Reply:
There should be some level of validation, but most of the validation you're doing in this function is extremely redundant.

Reply:
I think at this point, this is all we need:

```js
		if (side === '') {
			side = null;
		} if (typeof side === 'string') {
			side = this.sides[parseInt(side[1], 10)];
			if (!side) throw "Invalid side ID";
		}
```